tests/e2e/framework/kubernetes.go

```diff
@@ -138,11 +139,29 @@ func (k *KubeInfo) Teardown() error {
 	var err error
 	if k.namespaceCreated {
 		if err = util.DeleteNamespace(k.Namespace); err != nil {
-			glog.Error("Failed to delete namespace")
+			glog.Errorf("Failed to delete namespace %s", k.Namespace)
+			return err
+		}
+
+		// confirm the namespace is deleted as it will cause future creation to fail
+		maxAttempts := 15
```

Comment:
We have a retrier, maybe you would like to use this instead.
https://github.com/istio/istio/blob/da55d0a9832b402ef024df1e7a292185eb16c321/tests/e2e/util/kubeUtils.go#L84

Reply:
I don't see retrier as random ? this should probably have been changed

Reply:
ok i said random earlier as the comment said random in Backoff().  :)   if you look at the Backoff() code, it seems to pick a number that is based on baseDelay and backoffFactor (fixed). I don't see # of retries being used to influence the backoff.   https://github.com/istio/istio/blob/da55d0a9832b402ef024df1e7a292185eb16c321/tests/e2e/util/retry.go#L27